shotgun_api3/lib/mockgun.py

```diff
@@ -695,17 +740,21 @@ def _row_matches_filter(self, entity_type, row, sg_filter, retired_only):
             else:
                 raise ShotgunError("Unknown filter_operator type: %s" % operator)
         else:
+
             lval = self._get_field_from_row(entity_type, row, field)
 
             field_type = self._get_field_type(entity_type, field)
 
             # if we're operating on an entity, we'll need to grab the name from the lval's row
             if field_type == "entity":
-                lval_row = self._db[lval["type"]][lval["id"]]
-                if "name" in lval_row:
-                    lval["name"] = lval_row["name"]
-                elif "code" in lval_row:
-                    lval["name"] = lval_row["code"]
+                # If the entity field is set, we'll retrieve the name of the entity.
+                if lval is not None:
+                    lval_row = self._db[lval["type"]][lval["id"]]
```

Comment:
nitpick. difficult to parse visually. maybe break out the key string into a separate variable?